### ISSUE TITLE
Support for pilight 6.0

### ISSUE THREAD

**Issue by MrLeeh** [openhab/openhab#2301]
Since february 2015 pilight 6.0 has been released. There have been a lot of changes to version 5.0. In particular the structure of the configuration files has been simplified. There is now only one config file (`config.json`). Also device configuration and gui configuration have been separated. As a result of these changes there is no `location` parameter anymore when accessing the devices. The *openhab pilight plugin* can' t handle this as it still uses the location parameter. Is there a new version of the plugin that can handle pilight 6?

**Comment by fluidblue** [comment]
I would also like to use pilight v6 with the openhab pilight plugin..

**Comment by Oggerschummer** [comment]
+1 :-)
I was also happy to find pilight and the binding to openhab.
Unfortunately it´s no longer working with Version 6 of pilight.
Having a much cleaner configuration file is a major benefit and I´d 
be happy if the binding could reflect this.

**Comment by nightmare2105** [comment]
i used the executeCommandLine-Command as a walk-around:
```
rule "Lampe B cmd"
	when
		Item Lampe_B received command
	then 
		if (receivedCommand == OFF){
			executeCommandLine("/opt/openhab/pilight-switch.sh LampB off")
		}
		if (receivedCommand == ON){
			executeCommandLine("/opt/openhab/pilight-switch.sh LampB on")
		}
	end
```

with pilight-switch.sh:
```
#!/bin/bash
sudo pilight-control -d $1 -s $2 
```

unfortunately it's only one way, so any change outside of openhab can't be covered in openhab.
so i'm looking forward to an updated binding too.

**Comment by fluidblue** [comment]
I am using another work-around: The Exec binding instead of the pilight binding.
Here is how I define a switch:

    Switch Pollin_A "Pollin power outlet (A)" {exec=">[ON:/usr/local/bin/pilight-send --protocol=pollin --systemcode=31 --unitcode=1 --on] >[OFF:/usr/local/bin/pilight-send --protocol=pollin --systemcode=31 --unitcode=1 --off]"}

Unfortunately, both @nightmare2105 and my solution doesn't react on changes outside of openHAB. pilight has the cool feature that is listens with a 433MHz receiver to events from e.g. original remote controls. You can see this when using the pilight WebGUI: When using an original remote control, the pilight WebGUI updates the status.

In short: A pilight v6 binding is needed.

**Comment by teichsta** [comment]
@idserda could you please have a look at these requirements? Thanks, Thomas E.-E.

**Comment by idserda** [comment]
Code for this is almost ready, I'll try to submit a pull request this weekend.

**Comment by teichsta** [comment]
great, thanks for this update!

**Comment by MrLeeh** [comment]
@idserda Looking forward to this

**Comment by idserda** [comment]
See #2446

**Comment by idserda** [comment]
This issue can be closed since this has been implemented a long time ago :smiley:

**Comment by watou** [comment]
Thanks @idserda.
